From 2b0558c5811c7868e0a660f64d25a685c1f93c19 Mon Sep 17 00:00:00 2001
From: Maxime Chapelet <umxprime@videolabs.io>
Date: Tue, 19 Mar 2024 06:06:13 +0000
Subject: [PATCH 19/19] codec/videotoolbox: enable metal compatibility
 everywhere

* OpenGL compatibility only for macOS.
* OpenGLES compatibility everywhere else but visionOS

Given this, every decoded buffers will be compatible for all available
Darwin compatible vout renderers.
---
 modules/codec/videotoolbox/decoder.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/modules/codec/videotoolbox/decoder.c b/modules/codec/videotoolbox/decoder.c
index 2110b6296c..9cb729d6d6 100644
--- a/modules/codec/videotoolbox/decoder.c
+++ b/modules/codec/videotoolbox/decoder.c
@@ -1173,15 +1173,14 @@ static int StartVideoToolbox(decoder_t *p_dec)
         goto error;
     }
 
+    CFDictionarySetValue(destinationPixelBufferAttributes,
+                         kCVPixelBufferMetalCompatibilityKey,
+                         kCFBooleanTrue);
 #if TARGET_OS_OSX
    CFDictionarySetValue(destinationPixelBufferAttributes,
                         kCVPixelBufferIOSurfaceOpenGLTextureCompatibilityKey,
                         kCFBooleanTrue);
-#elif defined(TARGET_OS_VISION) && TARGET_OS_VISION
-    CFDictionarySetValue(destinationPixelBufferAttributes,
-                         kCVPixelBufferMetalCompatibilityKey,
-                         kCFBooleanTrue);
-#else // iOS and tvOS
+#elif !defined(TARGET_OS_VISION) || !TARGET_OS_VISION
     CFDictionarySetValue(destinationPixelBufferAttributes,
                          kCVPixelBufferOpenGLESCompatibilityKey,
                          kCFBooleanTrue);
-- 
2.32.1 (Apple Git-133)

